go-controller/pkg/ovn/master_test.go

```diff
@@ -1678,6 +1679,167 @@ var _ = ginkgo.Describe("Default network controller operations", func() {
 		})
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
+
+	ginkgo.It("verify IPs in host-network-namespace address_set after remote zone node add", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := config.InitConfig(ctx, nil, nil)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			config.OVNKubernetesFeature.EnableInterconnect = true
+			config.Kubernetes.HostNetworkNamespace = "ovn-host-network"
+
+			// PortBinding is required to add interconect resources for
+			// remote zone nodes
+			pb := &sbdb.PortBinding{
+				LogicalPort: "tstor-node1",
+				Datapath:    "tstor-node1",
+				TunnelKey:   5,
+			}
+			dbSetup.SBData = append(
+				dbSetup.SBData,
+				pb,
+			)
+
+			// Set annotations required to create remote zone node
+			newNodeSubnet := "10.1.2.0/24"
+			transitswitchSubnet := "100.88.0.3/16"
+			testNode.Annotations["k8s.ovn.org/network-ids"] = fmt.Sprintf("{\"default\":\"%d\"}", 0)
+			testNode.Annotations["k8s.ovn.org/node-chassis-id"] = "2"
+			testNode.Annotations["k8s.ovn.org/l3-gateway-config"] = `{"default":{"mac-address":"0a:58:01:02:03:04","ip-addresses":["1.2.3.4/24"],"ip-address":"1.2.3.4/24","next-hops":["1.2.3.1"],"next-hop":"1.2.3.1"}}`
+			testNode.Annotations["k8s.ovn.org/node-transit-switch-port-ifaddr"] = fmt.Sprintf("{\"ipv4\":\"%s\"}", transitswitchSubnet)
+			testNode.Annotations["k8s.ovn.org/node-mgmt-port-mac-address"] = "0a:58:0a:01:01:02"
+			testNode.Annotations["k8s.ovn.org/zone-name"] = "foo"
+			testNode.Annotations["k8s.ovn.org/node-gateway-router-lrp-ifaddr"] = ""
+			fakeOvn.startWithDBSetup(dbSetup,
+				&v1.NamespaceList{
+					Items: []v1.Namespace{*newNamespace(config.Kubernetes.HostNetworkNamespace)},
+				},
+
+				&v1.NodeList{
+					Items: []v1.Node{testNode},
+				},
+			)
+
+			// Required to make sure a remote zone node is being added
+			fakeOvn.controller.zone = "node1"
+
+			gomega.Expect(fakeOvn.controller.WatchNamespaces()).To(gomega.Succeed(), "Namespace should be created fine")
+			err = fakeOvn.controller.WatchNodes()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			// Set annotations required to create remote zone node
+			nodeAnnotator := kube.NewNodeAnnotator(&kube.Kube{KClient: fakeOvn.fakeClient.KubeClient}, testNode.Name)
+			err = util.SetNodeHostSubnetAnnotation(nodeAnnotator, ovntest.MustParseIPNets(newNodeSubnet))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			nodeAnnotator.Set("k8s.ovn.org/node-gateway-router-lrp-ifaddr", "{\"ipv4\":\"100.64.0.3/16\"}")
+			err = nodeAnnotator.Run()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			// Get updated node after setting all required annotations for
+			// remote zone node and allow enough time to create interconnect
+			// resources and HostNetworkNamespace address_set to be updated
+			updatedNode, _ := fakeOvn.fakeClient.KubeClient.CoreV1().Nodes().Get(context.TODO(), testNode.Name, metav1.GetOptions{})
+			time.Sleep(time.Second)
```

Comment:
why are we sleeping here? if its to wait for annotations, the correct approach is to use gomega.Eventually to check for the annotation patch and then get that. So that whole block below this should go into an eventually construct (see other test examples)
adding random sleeps in tests without justifications is not a good practice

Reply:
I put a sleep here to give controller enough time to add node IPs to HostNetworkNamespace address set. However 1 second seems too much to me now, i have reduced it to 100 millisecond. Is this acceptable?

Reply:
stopped using sleep after discussing with Surya.